cookbooks/bcpc/recipes/powerdns.rb

```diff
@@ -45,259 +44,214 @@
 subnet = node[:bcpc][:management][:subnet]
 raise "Did not get a subnet" if not subnet
 
-%w{pdns-server pdns-backend-mysql}.each do |pkg|
-    package pkg do
-        action :upgrade
-    end
-end
+if node[:bcpc][:management][:vip] and get_nodes_for("mysql").length() > 0
+#  make_config('mysql-pdns-user', "pdns")
+#  make_config('mysql-pdns-password', secure_password)
+
+  allnodes = Array.new
```

Comment:
The code for `get_node_attributes` (and here since you want all nodes it's [`get_req_node_attributes`](https://github.com/bloomberg/chef-bach/blob/master/cookbooks/bcpc/libraries/utils.rb#L216-237)) you would use is:
```
MY_ATTR_SRCH_KEYS = {
  'hostname' => 'hostname',
  'management_ip' => 'bcpc.management.ip',
  'storage_ip' => 'bcpc.storage.ip', 
  'floating_ip' => 'bcpc.floating.ip'}
allnodes.push(get_req_node_attributes(MY_ATTR_SRCH_KEYS))
```